### PR TITLE
Screen docs

### PR DESCRIPTION
**Pull request by willmcgugan**
Add docs for the new screen callbacks

### CODE REVIEW

File: docs/guide/screens.md [hunk 1]
```diff
@@ -219,3 +219,40 @@ Let's see what happens when we use `ModalScreen`.
...
+    --8<-- "docs/examples/guide/screens/modal01.css"
+    ```
+
+In the `on_button_pressed` message handler we call `dismiss` with a boolean that indicates if the user has chosen to quit the app.
+This boolean is passed to the `check_quit` function we provided when `QuitScreen` was pushed.
+
+Although this example behaves the same as the previous code, it is more flexible because it has removed responsibility for exiting from the modal screen to the caller.
+This makes it easier for the app to perform any cleanup actions prior to exiting, for example.
+
+Returning data in this way can help keep your code manageable by making it easy to re-use your Screen classes in other contexts.
```
Comment (davep) [inline docs/guide/screens.md:253]:
```suggestion
Returning data in this way can help keep your code manageable by making it easy to re-use your `Screen` classes in other contexts.
```

File: docs/examples/guide/screens/modal03.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    def on_button_pressed(self, event: Button.Pressed) -> None:
+        if event.button.id == "quit":
+            self.dismiss(True)
+        else:
+            self.dismiss(False)
+
+
+class ModalApp(App):
+    """A app with a modal dialog."""
```
Comment (davep) [inline docs/examples/guide/screens/modal03.py:34]:
```suggestion
    """An app with a modal dialog."""
```

File: docs/examples/guide/screens/modal03.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    CSS_PATH = "modal01.css"
+    BINDINGS = [("q", "request_quit", "Quit")]
+
+    def compose(self) -> ComposeResult:
+        yield Header()
+        yield Label(TEXT * 8)
+        yield Footer()
+
+    def action_request_quit(self) -> None:
+        """Action to display quit dialog."""
```
Comment (davep) [inline docs/examples/guide/screens/modal03.py:45]:
```suggestion
        """Action to display quit the dialog."""
```

File: docs/examples/guide/screens/modal03.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        def check_quit(quit: bool) -> None:
+            """Called when QuitScreen is dismissed."""
```
Comment (davep) [inline docs/examples/guide/screens/modal03.py:48]:
```suggestion
            """Called when `QuitScreen` is dismissed."""
```

File: docs/examples/guide/screens/modal02.py [hunk 5]
```diff
@@ -42,6 +42,7 @@ def compose(self) -> ComposeResult:
         yield Footer()
 
     def action_request_quit(self) -> None:
+        """Action to display quit dialog."""
```
Comment (davep) [inline docs/examples/guide/screens/modal02.py:45]:
```suggestion
        """Action to display the quit dialog."""
```

**Review by davep — APPROVED** [review]
"I Rodrigoed that PR"

File: docs/examples/guide/screens/modal03.py (hunk 3 above)
Comment (willmcgugan) [inline docs/examples/guide/screens/modal03.py:45]:
```suggestion
        """Action to display the quit dialog."""
```

**Comment by willmcgugan** [comment]
We should get swag with "You've been Rodrigoed" printed on it.

**Comment by davep** [comment]
Should be a good old rubber stamp.

**Comment by rodrigogiraoserrao** [comment]
> We should get swag with "You've been Rodrigoed" printed on it.

I'll take those to EP

**Pull request merged**